### ISSUE TITLE
Journal search algorithm / Subscription identifier not functioning as hoped

### ISSUE THREAD

**Issue by JosephMcArthur** [OAButton/discussion#531]
This was a known bug on delivery, however, we have more test cases now. Clearly we need a route to handle this.

> Nicholas et al 2016 doi:10.1002/leap.1087 We have a sub but the OAB drops to "Request an ILL" in red)
> Britz 2010 10.1177/0165551510382071 We have a sub but the OAB drops to to "Request an ILL" in red.
> Al-Jawad et al. 2004 10.1016/j.physb.2004.03.150 We have a sub but the OAB drops to to "Request an ILL" in red.
> Simmons et al. 2011 10.1111/j.1600-0722.2011.00909.x/abstract We have a sub but the OAB drops to to "Request an ILL" in red.
> Cohen 2010 doi:10.1177/0170840609357381 We have a sub and the OAB finds this, with a link to login.
> Ahmed 2016 10.1177/1363460716629607 There is a problem with the flow: we have a subscription, but OAB finds a metadata-only record in the Cardiff repo in preference: http://orca.cf.ac.uk/80272/

**Comment by markmacgillivray** [comment]
I have improved our search of Primo.

For the first one the journal is now found in primo search but does not have any info about where it is in the library. 

The assumption from the journal Science and its result in Primo is that if it has a "library" object then it is in Imperial library - e.g they hold copies somewhere. This field does not appear for Learned Publishing - the journal of the first article. It may well be the case that if we can find the journal in Primo at all, even with no info about it being IN the library, that this indicates Imperial have a subscription but just don't hold any of it on site. If this can be confirmed with Imperial, and if they happen to know the name of a journal they don't subscribe to, I can confirm this.

For now, I have made this assumption that if I can find it in Primo then Imperial DO subscribe to it. Doing this, I confirm showing successful "available in library links" for all of the above.

Assigning to @JosephMcArthur for info for the followup email
